CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+# Contributing to B2 Command Line Tool
+
+We encourage outside contributors to perform changes on our codebase. Many such changes have been merged already. In order to make it easier to contribute, core developers of this project:
+
+* provide guidance (through the issue reporting system)
+* provide tool assisted code review (through the Pull Request system)
+* maintain a set of unit tests
+* maintain a set of integration tests (run with a production cloud)
+* maintain development automation tools using [nox](https://github.com/theacodes/nox) that can easily:
+   * format the code using [yapf](https://github.com/google/yapf)
+   * runs linters to find subtle/potential issues with maintainability
+   * run the test suite on multiple Python versions using [pytest](https://github.com/pytest-dev/pytest)
+* maintain Continuous Integration (by using GitHub Actions) that:
+   * runs all sorts of linters
+   * checks if the Python distribution can be built
+   * runs all tests on a matrix of 6 versions of Python (including pypy) and 3 operating systems (Linux, Mac OS X and Windows)
+   * checks if the documentation can be built properly
+* maintain other Continuous Integration tools (coverage tracker)
+
+## Developer Info
+
+You'll need to have [nox](https://github.com/theacodes/nox) installed:
+
+* `pip install nox`
+
+With `nox`, you can run different sessions (default are `lint` and `test`):
```

Comment:
I'm always implementing default behavior of nox in a way, that it doesn't change the source code as `format` does - to simply check the code quality before commit/push. Running `lint` before `format` doesn't make any sense as every tool used in format (currently only yapf) has also "lining" feature used in `lint`

Reply:
Regarding the incremental formatter, non-incremental one is not that slow (in parallel), and I don't really like assuming the current state of branches like in the old implementation. We work with forks and different remotes. I would never use it personally, but I can agree that it might be helpful for someone. Maybe we should implement incremental formatter to format only files not yet committed? I know that black has this feature.

Reply:
you can run tests every 3 commits and then it wouldn't catch your changes.

The incremental formatter worked quite well so far, I don't understand why you wouldn't want to use it? Maybe the current code formats way faster than the one we used to have when I wrote that incremental behavior, it was unusable in a full run back then.

Reply:
How long does this take now? It used to be >40s for the entire thing regardless of number of cores (some files were just processed very slowly)

Reply:
On my hardware:

CLI:
~1.5s on max 8 cores
~3.5s on 1 core

SDK:
~4.5s on max 8 cores
~12s on 1 core

Reply:
that is way different from what we used to see, we must have changed the problematic places in the code so that it's now way easier for yapf to calculate it. Still, I can't see why we can't use the incremental thing (since we already have the code for it)?

Reply:
after discussion, its not really worth to port it when it's that fast